src/vscode-dts/vscode.proposed.notebookExecution.d.ts

```diff
@@ -0,0 +1,39 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+declare module 'vscode' {
+
+	/**
+	 * A NotebookExecution is how {@link NotebookController notebook controller} can indicate whether the notebook controller is busy or not.
+	 *
+	 * When {@linkcode NotebookExecution.start start()} is called on the execution task, it causes the Notebook to enter a executing state .
+	 * When {@linkcode NotebookExecution.end end()} is called, it enters the idle state.
+	 */
+	export interface NotebookExecution {
```

Comment:
I suppose normally I'd say this should start automatically when you create it, and then be disposable, but we should probably use start/end just to match the cell execution pattern.

Reply:
Yup, thats the pattern is tried to stick with, 
I thought of just returning a `IDisposable` instead of `NotebookExecution`, that was much simpler, but didn't seem to line up with the cell exec code.